Module#04/web-app/src/main/java/com/epam/esm/controller/AuthenticationController.java

```diff
@@ -0,0 +1,80 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.security.auth.AuthenticationRequest;
+import com.epam.esm.security.auth.AuthenticationResponse;
+import com.epam.esm.security.auth.RegisterRequest;
+import com.epam.esm.security.service.AuthenticationService;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Controller class for handling authentication-related requests.
+ * <p>
+ * This class is responsible for handling registration and authentication requests.
+ */
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping(value = "/token")
+public class AuthenticationController {
+    private final AuthenticationService service;
+
+    /**
+     * Endpoint for user registration.
+     *
+     * @param request The registration request containing user details.
+     * @return ResponseEntity containing the authentication response.
+     */
+    @PostMapping("/register")
+    public ResponseEntity<AuthenticationResponse> register(
+            final @RequestBody RegisterRequest request) {
+        return ResponseEntity.ok(service.register(request));
+    }
+
+    /**
+     * Endpoint for user authentication.
+     *
+     * @param request The authentication request containing user credentials.
+     * @return ResponseEntity containing the authentication response.
+     */
+    @PostMapping("/authenticate")
+    public ResponseEntity<AuthenticationResponse> authenticate(
+            final @RequestBody AuthenticationRequest request) {
+        return ResponseEntity.ok(service.authenticate(request));
+    }
+
+    /**
+     * Refreshes the access token and returns a new AuthenticationResponse.
+     *
+     * @param request  The HttpServletRequest object.
+     * @param response The HttpServletResponse object.
+     * @return ResponseEntity containing the refreshed AuthenticationResponse.
+     */
+    @PostMapping("/refresh")
+    public ResponseEntity<AuthenticationResponse> refreshTokens(
+            final HttpServletRequest request,
```

Comment:
HttpServletRequest is overhead to be used. it is better to use @Header("Authorization") annotation

Reply:
> HttpServletRequest is overhead to be used. it is better to use @Header("Authorization") annotation

> modified signature refreshTokens to @RequestHeader("Authorization") String authorizationHeader